common/lib/xmodule/xmodule/capa_module.py

```diff
@@ -144,17 +144,19 @@ def handle_ajax(self, dispatch, data):
         """
         # self.score is initialized in self.lcp but in this method is accessed before self.lcp so just call it first.
         self.lcp
-        handlers = {
-            'hint_button': self.hint_button,
-            'problem_get': self.get_problem,
-            'problem_check': self.submit_problem,
-            'problem_reset': self.reset_problem,
-            'problem_save': self.save_problem,
-            'problem_show': self.get_answer,
-            'score_update': self.update_score,
-            'input_ajax': self.handle_input_ajax,
-            'ungraded_response': self.handle_ungraded_response
-        }
+        handlers = {}
+        if self.scope_ids.user_id:
```

Comment:
Seems like these handlers were intentionally allowed to un-authenticated users third part service-to-service calls, see https://github.com/edx/edx-platform/pull/1791 for more.

I believe you need to dig into it to find out a proper solution for this.